src/depackers/uncompress.c

```diff
@@ -220,23 +232,25 @@ static int decrunch_compress(HIO_HANDLE * in, FILE * out, long inlen)
 
 			/* And put them out in forward order */
 
-			if (outpos + (i = (de_stack - stackp)) >= OBUFSIZ) {
+			if (outpos + (i = (de_stack - stackp)) >= outsize) {
 				do {
-					if (i > OBUFSIZ - outpos)
-						i = OBUFSIZ - outpos;
+					if (i > outsize - outpos)
+						i = outsize - outpos;
 
 					if (i > 0) {
 						memcpy(outbuf + outpos, stackp, i);
 						outpos += i;
 					}
 
-					if (outpos >= OBUFSIZ) {
-						if (fwrite(outbuf, 1, outpos, out) != outpos) {
+					if (outpos >= outsize) {
+						outsize += OBUFSIZ;
+
+						tmp = (char_type *) realloc(outbuf, outsize + 2048);
```

Comment:
Pretty much all of the streaming depackers need a slightly different approach here to avoid slow loads and the Linux OOM killer. Instead of expanding the output buffer linearly, it should be expanded exponentially and bounded against `LIBXMP_DEPACK_LIMIT`. Something like this *might* work (there's a `next_power_of_two` in bunzip2.c that can be copied):

```C
...
if (outpos >= outsize) {
  outsize = next_power_of_two(outpos);
  if (outsize > LIBXMP_DEPACK_LIMIT) {
    free(outbuf);
    return -1;
  }
  tmp = (char_type *) realloc(outbuf, outsize + 2048);
  ...
```

Reply:
I'll check my fuzzing corpus for any existing files that generate slow loads with this patch just in case.

Reply:
Nope, nothing came up. I think this can just be left as-is for now, I want to see if libFuzzer turns up anything like it did for the others.